utils/custom_logger.py

```diff
@@ -0,0 +1,25 @@
+"""Custom logging module with some formatting."""
+
+import sys
+import logging
+
+
+def setup_logger(name: str) -> logging.Logger:
```

Comment:
Would it be better to use the logging dictconfig/yaml like so ? And then one can simply use `logger = getLogger(__name__)` when/wherever they need it. Do make changes to the dictconfig as required if you wanted to implement it.

```python

import logging
import logging.config

logging_config = {
    'version': 1,
    'formatters': {
        'standard': {
            'format': '%(asctime)s - %(name)s - %(levelname)s - %(message)s'
        }
    },
    'handlers': {
        'file': {
            'class': 'logging.FileHandler',
            'filename': 'app.log',
            'level': 'DEBUG',
            'formatter': 'standard'
        },
        'console': {
            'class': 'logging.StreamHandler',
            'level': 'INFO',
            'formatter': 'standard',
            'stream': 'ext://sys.stdout'
        }
    },
    'loggers': {
        '': {
            'handlers': ['file', 'console'],
            'level': 'DEBUG',
            'propagate': True
        }
    }
}

logging.config.dictConfig(logging_config) #configure it during the startup

# Example usage
logger = logging.getLogger(__name__)
logger.debug('This is a debug message')
logger.info('This is an info message')
logger.warning('This is a warning message')
logger.error('This is an error message')
logger.critical('This is a critical message')